gradle-conjure/src/main/java/com/palantir/gradle/conjure/GradleExecUtils.java

```diff
@@ -16,130 +16,30 @@
-
-        if (execResult.getExitValue() != 0) {
-            throw new RuntimeException(String.format(
-                    "Failed to %s. The command '%s' failed with exit code %d. Output:\n%s",
-                    failedTo, combinedArgs, execResult.getExitValue(), output.toString()));
-        }
-    }
-
-    private static void runJavaCodeInProcess(String failedTo, List<String> combinedArgs, Method mainMethod) {
-        try {
-            String[] args = combinedArgs.toArray(new String[] {});
-            mainMethod.invoke(null, new Object[] {args});
-        } catch (Throwable t) {
-            Throwable rootCause = Throwables.getRootCause(t);
-            if (rootCause instanceof GradleExecStubs.ExitInvoked) {
-                int exitStatus = ((GradleExecStubs.ExitInvoked) rootCause).getExitStatus();
-                if (exitStatus == 0) {
-                    // Exit status zero, we're good to go!
-                    return;
-                }
-                // the error message from a generator attempting to call exit 1 looks pretty gross
-                throw new RuntimeException(String.format(
-                        "Failed to %s. The command '%s' failed with exit code 1. Output above.",
-                        failedTo, combinedArgs));
-            }
-            throw new RuntimeException(
-                    String.format("Failed to %s. The command '%s' failed.", failedTo, combinedArgs), t);
-        }
-    }
-
-    private static Optional<Method> getMainMethod(URLClassLoader classLoader, String mainClassName) {
-        try {
-            ClassFileLocator locator = new ClassFileLocator.ForUrl(classLoader.getURLs());
-            TypePool typePool = TypePool.ClassLoading.of(classLoader);
-            Class<?> mainClass = new ByteBuddy(ClassFileVersion.ofThisVm(ClassFileVersion.JAVA_V8))
-                    .redefine(typePool.describe(mainClassName).resolve(), locator)
-                    .name(mainClassName + "RedefinedForGradleConjure")
-                    .visit(new ForDeclaredMethods()
-                            .invokable(
-                                    ElementMatchers.any(),
-                                    MemberSubstitution.relaxed()
-                                            .method(ElementMatchers.is(System.class.getMethod("exit", int.class)))
-                                            .replaceWith(GradleExecStubs.getStubMethod())))
-                    .make(typePool)
-                    .load(classLoader, ClassLoadingStrategy.Default.INJECTION)
-                    .getLoaded();
-
-            return Optional.of(mainClass.getMethod("main", String[].class));
-        } catch (ReflectiveOperationException e) {
-            log.warn("Failed too get main method {}", mainClassName, e);
-            return Optional.empty();
-        }
+        project.getGradle()
+                .getSharedServices()
+                .registerIfAbsent(
+                        "conjure-runner-" + executable,
```

Comment:
maybe this is what we were slacking about, but if you didn't have this `executable` concatenated in the name here, do all tasks try to use the same SharedService???

Reply:
Yep. I updated this with a comment a few minutes ago. The `registerIfAbsent` method _only_ takes the name into account, not even the service type. Initially my tests failed attempting to run the conjure compiler with args to generate objects...